src/main/java/uk/nhs/digital/nhsconnect/lab/results/inbound/EdifactParser.java

```diff
@@ -68,47 +66,19 @@ private Interchange parseInterchange(List<String> allEdifactSegments) {
 
     private Message parseMessage(List<String> singleMessageEdifactSegments) {
         var messageTrailerIndex = singleMessageEdifactSegments.size() - 1;
-        var firstTransactionStartIndex = findAllIndexesOfSegment(singleMessageEdifactSegments, TRANSACTION_START_SEGMENT).stream()
+        var firstMessageStartIndex = findAllIndexesOfSegment(singleMessageEdifactSegments, MESSAGE_END_SEGMENT).stream()
                 .findFirst()
-                // there might be no transaction inside - RECEP - so all message lines belong to the message
                 .orElse(messageTrailerIndex);
 
-        // first lines until transaction
-        var onlyMessageLines = new ArrayList<>(singleMessageEdifactSegments.subList(0, firstTransactionStartIndex));
+        // first lines until end of message
+        var onlyMessageLines = new ArrayList<>(singleMessageEdifactSegments.subList(0, firstMessageStartIndex));
         onlyMessageLines.add(singleMessageEdifactSegments.get(messageTrailerIndex));
 
         var message = new Message(onlyMessageLines);
-        var transactions = parseAllTransactions(singleMessageEdifactSegments);
-        transactions.forEach(transaction -> transaction.setMessage(message));
-        message.setTransactions(transactions);
 
         return message;
```

Comment:
No need for variable before return anymore.
```suggestion
        return new Message(onlyMessageLines);
```
It shows up in my IDE with a warning, but I'm aware some people like it because debugging is a *little* easier.

Reply:
Have done this in 697ebb8